### DISCUSSION TITLE
Unable to get to print list.

### DISCUSSION THREAD

**Discussion by virtualstorm24** [community/community#64819] [Programming Help]
### Body

I am trying to print a list of the names and ages from the 9/11 casualties from three separate wiki links. 
There are no errors being thrown out, but It's not printing out the list.

`from bs4 import BeautifulSoup as bs
import requests

urls = [
    "https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks_(A%E2%80%93G)_",
    "https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks_(H%E2%80%93N)_",
    "https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks_(O%E2%80%93Z)_"
]

data = []

for url in urls:
    response = requests.get(url)
    soup = bs(response.content, 'html.parser')

    # table = soup.select(".wikitable.sortable")
    table = soup.find_all(class_="wikitable")

    if table:
        rows = table[0].find_all('tr')
        for row in rows[2:]:
            cells = row.find_all('td')
            if len(cells) == 2:
                name = cells[0].get_text(strip=True)
                age = cells[1].get_text(strip=True)
                data.append([name, age])
                # print("Name:", name)
                # print("Age:", age)

    else:
        print("No table found", url)

for name, _ in data:
    print(name)

`

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by arafajs** [comment]
You are using the `select()` method to find the table, but this method only returns the first matching element. You need to use the `find_all() `method instead, which will return a list of all matching elements.

Here is the updated code:

```
from bs4 import BeautifulSoup as bs
import requests

urls = [
"https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks_(A%E2%80%93G)",
"https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks(H%E2%80%93N)",
"https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks(O%E2%80%93Z)_"
]

data = []

for url in urls:
response = requests.get(url)
soup = bs(response.content, 'html.parser')

table = soup.find_all(class_="wikitable")

if table:
    rows = table[0].find_all('tr')
    for row in rows[2:]:
        cells = row.find_all('td')
        if len(cells) == 2:
            name = cells[0].get_text(strip=True)
            age = cells[1].get_text(strip=True)
            data.append([name, age])

else:
    print("No table found", url)

for name, _ in data:
print(name)
```

**Comment by nftushar** [comment]
It seems like the indentation in your code is not correct, which can lead to issues in Python. Also, there are a few places where you need to adjust your code to properly extract the data from the Wikipedia pages. Here's the corrected version of your code:

python
Copy code
from bs4 import BeautifulSoup as bs
import requests

urls = [
    "https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks_(A%E2%80%93G)",
    "https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks(H%E2%80%93N)",
    "https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks(O%E2%80%93Z)_"
]

data = []

for url in urls:
    response = requests.get(url)
    soup = bs(response.content, 'html.parser')

    # Find the correct table with the data
    table = soup.find(class_="wikitable")
    
    if table:
        rows = table.find_all('tr')
        for row in rows[1:]:  # Start from index 1 to skip the header row
            cells = row.find_all('td')
            if len(cells) == 2:
                name = cells[0].get_text(strip=True)
                age = cells[1].get_text(strip=True)
                data.append([name, age])
    else:
        print("No table found at", url)

# Print the extracted names
for name, _ in data:
    print(name)
Key points of correction in the code:

Indentation: The code within the loop and conditional statements should be properly indented.

Table Selection: In the code, I used table = soup.find(class_="wikitable") to find the table containing the data. This assumes that the class "wikitable" is assigned to the correct table. If this doesn't work, you might need to adjust the CSS class selector accordingly based on the actual structure of the Wikipedia page.

Rows Iteration: I adjusted the loop to start from index 1 (rows[1:]) to skip the header row of the table, as the header row doesn't contain the actual data.

Printing Names: The corrected code will print the extracted names.

Please try running this corrected code and see if it extracts and prints the names from the Wikipedia pages as expected.

**Comment by nirav-gajera** [comment]
try:

```
from bs4 import BeautifulSoup as bs
import requests

urls = [
"https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks_(A%E2%80%93G)",
"https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks(H%E2%80%93N)",
"https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks(O%E2%80%93Z)_"
]

data = []

for url in urls:
response = requests.get(url)
soup = bs(response.content, 'html.parser')

table = soup.find_all(class_="wikitable")

if table:
    rows = table[0].find_all('tr')
    for row in rows[2:]:
        cells = row.find_all('td')
        if len(cells) == 2:
            name = cells[0].get_text(strip=True)
            age = cells[1].get_text(strip=True)
            data.append([name, age])
            # print("Name:", name)
            # print("Age:", age)

else:
    print("No table found", url)
for name, _ in data:
print(name)

```

**Comment by virtualstorm24** [comment]
I figured it out.

`from bs4 import BeautifulSoup as bs
import requests

urls = [
    "https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks_(A%E2%80%93G)",
    "https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks_(H%E2%80%93N)",
    "https://en.wikipedia.org/wiki/List_of_victims_of_the_September_11_attacks_(O%E2%80%93Z)"
]

data = []

for url in urls:
    r = requests.get(url)
    soup = bs(r.content, 'html.parser')

    table = soup.find_all(class_="wikitable")

    if table:
        rows = table[0].find_all('tr')
        for row in rows[2:]:
            cells = row.find_all('td')
            if len(cells) >= 2:
                name_cell = cells[0]
                age_cell = cells[1]
                name = name_cell.get_text(strip=True)
                age = age_cell.get_text(strip=True)
                data.append([name, age])

for name, age in data:
    print("Name:", name)
    print("Age:", age)
    print()`

**Reply by arafajs** [reply]
@virtualstorm24 That's great!
